src/PointwiseFunctions/AnalyticData/NewtonianEuler/SodExplosion.cpp

```diff
@@ -0,0 +1,137 @@
+#include "Utilities/Math.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+/// \cond
+namespace NewtonianEuler::AnalyticData {
+template <size_t Dim>
+SodExplosion<Dim>::SodExplosion(const double initial_radius,
+                                const double inner_mass_density,
+                                const double inner_velocity,
+                                const double inner_pressure,
+                                const double outer_mass_density,
+                                const double outer_velocity,
+                                const double outer_pressure) noexcept
+    : initial_radius_(initial_radius),
+      inner_mass_density_(inner_mass_density),
+      inner_velocity_(inner_velocity),
+      inner_pressure_(inner_pressure),
+      outer_mass_density_(outer_mass_density),
+      outer_velocity_(outer_velocity),
+      outer_pressure_(outer_pressure),
+      equation_of_state_(1.4) {}
+
+template <size_t Dim>
+void SodExplosion<Dim>::pup(PUP::er& p) noexcept {
+  p | initial_radius_;
+  p | inner_mass_density_;
+  p | inner_velocity_;
+  p | inner_pressure_;
+  p | outer_mass_density_;
+  p | outer_velocity_;
+  p | outer_pressure_;
+  p | equation_of_state_;
+}
+
+template <size_t Dim>
+bool operator==(const SodExplosion<Dim>& lhs,
+                const SodExplosion<Dim>& rhs) noexcept {
+  return lhs.initial_radius_ == rhs.initial_radius_ and
+         lhs.inner_mass_density_ == rhs.inner_mass_density_ and
+         lhs.inner_velocity_ == rhs.inner_velocity_ and
+         lhs.inner_pressure_ == rhs.inner_pressure_ and
+         lhs.outer_mass_density_ == rhs.outer_mass_density_ and
+         lhs.outer_velocity_ == rhs.outer_velocity_ and
+         lhs.outer_pressure_ == rhs.outer_pressure_;
+}
+
+template <size_t Dim>
+bool operator!=(const SodExplosion<Dim>& lhs,
+                const SodExplosion<Dim>& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+template <size_t Dim>
+tuples::TaggedTuple<Tags::MassDensity<DataVector>> SodExplosion<Dim>::variables(
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& x,
+    tmpl::list<Tags::MassDensity<DataVector>> /*meta*/) const noexcept {
+  auto mass_density = make_with_value<Scalar<DataVector>>(x, 0.0);
+  std::array<double, Dim> coords{};
+  for (size_t s = 0; s < get_size(get<0>(x)); ++s) {
+    for (size_t d = 0; d < Dim; ++d) {
+      gsl::at(coords, d) = get_element(x.get(d), s);
+    }
+    const double radius = magnitude(coords);
+    get_element(get(mass_density), s) =
+        radius > initial_radius_ ? outer_mass_density_ : inner_mass_density_;
+  }
+  return mass_density;
+}
+
+template <size_t Dim>
+tuples::TaggedTuple<Tags::Velocity<DataVector, Dim, Frame::Inertial>>
+SodExplosion<Dim>::variables(
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& x,
+    tmpl::list<Tags::Velocity<DataVector, Dim, Frame::Inertial>> /*meta*/)
+    const noexcept {
+  auto velocity = make_with_value<tnsr::I<DataVector, Dim, Frame::Inertial>>(
+      get<0>(x), 0.0);
```

Comment:
weeeeellllllll if you're going to hardcode the velocity to 0, why do you have options and member variables for it? 😛